src/commonMain/kotlin/com/charleskorn/kaml/Yaml.kt

```diff
@@ -18,20 +18,111 @@
 ) : StringFormat {
-    public val configuration: YamlConfiguration
-
-    public fun <T> decodeFromYamlNode(deserializer: DeserializationStrategy<T>, node: YamlNode): T
 
     public companion object {
-        public val default: Yaml
+        public val default: Yaml = Yaml()
+    }
+
+    public fun <T> decodeFromYamlNode(
+        deserializer: DeserializationStrategy<T>,
+        node: YamlNode,
+    ): T {
+        val input = YamlInput.createFor(node, this, serializersModule, configuration, deserializer.descriptor)
+        return input.decodeSerializableValue(deserializer)
+    }
+
+    override fun <T> decodeFromString(
+        deserializer: DeserializationStrategy<T>,
+        string: String,
+    ): T {
+        return decodeFromSource(deserializer, string.bufferedSource())
+    }
+
+    public fun <T> decodeFromSource(
+        deserializer: DeserializationStrategy<T>,
+        source: Source,
+    ): T {
+        val rootNode = parseToYamlNode(source)
+
+        val input = YamlInput.createFor(rootNode, this, serializersModule, configuration, deserializer.descriptor)
+        return input.decodeSerializableValue(deserializer)
+    }
+
+    public fun parseToYamlNode(string: String): YamlNode =
+        parseToYamlNode(string.bufferedSource())
+
+    internal fun parseToYamlNode(source: Source): YamlNode {
+        val parser = YamlParser(source)
+        val reader =
+            YamlNodeReader(parser, configuration.extensionDefinitionPrefix, configuration.allowAnchorsAndAliases)
+        val node = reader.read()
+        parser.ensureEndOfStreamReached()
+        return node
+    }
+
+    public fun <T> encodeToSink(
+        serializer: SerializationStrategy<T>,
+        value: T,
+        sink: Sink,
+    ) {
+        encodeToBufferedSink(serializer, value, sink.buffer())
+    }
+
+    override fun <T> encodeToString(
+        serializer: SerializationStrategy<T>,
+        value: T,
+    ): String {
+        val buffer = Buffer()
+        encodeToBufferedSink(serializer, value, buffer)
+        return buffer.readUtf8().trimEnd()
+    }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    private fun <T> encodeToBufferedSink(
+        serializer: SerializationStrategy<T>,
+        value: T,
+        sink: BufferedSink,
+    ) {
+        BufferedSinkDataWriter(sink).use { writer ->
+            YamlOutput(writer, serializersModule, configuration).use { output ->
+                output.encodeSerializableValue(serializer, value)
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalStdlibApi::class)
+private class BufferedSinkDataWriter(
+    val sink: BufferedSink,
+) : StreamDataWriter, AutoCloseable {
+    override fun flush(): Unit = sink.flush()
+
+    override fun write(str: String) {
+        sink.writeUtf8(str)
+    }
+
+    override fun write(str: String, off: Int, len: Int) {
+        sink.writeUtf8(string = str, beginIndex = off, endIndex = off + len)
+    }
+
+    override fun close() {
+        flush()
     }
```

Comment:
I'm not 100% certain flushing is necessary or important, but I'm concerned about missing data in resources-intensive situations. It probably doesn't hurt.